cmd/commands/utils.go

```diff
@@ -21,16 +23,87 @@ import (
 	tmtypes "github.com/tendermint/tendermint/types"
 )
 
+//This variable can be overwritten by plugin applications
+// if they require a different working directory
+var DefaultHome = "basecoin"
+
 func BasecoinRoot(rootDir string) string {
 	if rootDir == "" {
 		rootDir = os.Getenv("BCHOME")
 	}
 	if rootDir == "" {
-		rootDir = os.Getenv("HOME") + "/.basecoin"
+		rootDir = os.Getenv("HOME") + "/." + DefaultHome
 	}
 	return rootDir
 }
 
+//Add debugging flag and execute the root command
+func ExecuteWithDebug(RootCmd *cobra.Command) {
+
+	var debug bool
+	RootCmd.PersistentFlags().BoolVar(&debug, "debug", false, "enables stack trace error messages")
+
+	//note that Execute() prints the error if encountered, so no need to reprint the error,
```

Comment:
This is bad, because it also prints the help menu, which may be totally irrelevant.